src/github.rs

```diff
@@ -990,6 +993,8 @@ pub struct IssuesEvent {
     pub repository: Repository,
     /// The GitHub user that triggered the event.
     pub sender: User,
+    /// Assignee affected by this `action`
+    pub assignee: User,
```

Comment:
This isn't right. Not every action has an `assignee`. This should be under `Assigned` and `Unassigned` in `IssuesAction`.

Reply:
I'm sorry, I'm not sure i understand this comment: what do you mean that this `assignee` should go in `IssuesAction`? That's the enum describing all possible actions, right?

My understanding was: when the action is `Assigned` or `Unassigned`, then the `issue.assignee` field is the user object of the action (either they are unassigned the PR or become the new assignee).

Maybe `assignee: Option<User>` since it can be null?

thanks :slightly_smiling_face:

Reply:
For example, the `Labeled` and `Unlabeled` variants have a `label` field, because `label` is only present on those actions.

Similarly, `assignee` is only present on `Assigned` and `Unassigned` actions.

`assignee: Option<User>` could work, but is not appropriate; for other actions, we never expect assignee, for example. And for assigned and unassigned we always expect it.

Reply:
ok @jackh726 I've refactored the code as per your suggestion, seems to work.

As an aside the code now looks a bit strange because of this handling of enums variants with structs inside. I have to squint to understand what it does (and I wrote it). Any suggestion on how to improve the readability?